packages/ts/react-signals/test/FullStackSignal.spec.tsx

```diff
@@ -4,13 +4,50 @@ import { render } from '@testing-library/react';
 import { ConnectClient, type Subscription } from '@vaadin/hilla-frontend';
 import sinon from 'sinon';
 import sinonChai from 'sinon-chai';
-import { type StateEvent, StateEventType } from '../src/FullStackSignal.js';
-import { NumberSignal } from '../src/index.js';
+import { DependencyTrackingSignal, type StateEvent, StateEventType } from '../src/FullStackSignal.js';
+import { computed, NumberSignal } from '../src/index.js';
 import { nextFrame } from './utils.js';
 
 use(sinonChai);
 
 describe('@vaadin/hilla-react-signals', () => {
+  describe('DependencyTrackingSignal', () => {
+    class TestSignal<T = unknown> extends DependencyTrackingSignal<T> {
+      constructor(value: T | undefined, onSubscribe: () => void, onUnsubscribe: () => void) {
+        super(value, onSubscribe, onUnsubscribe);
+        this.subscribe(() => {}); // Ignores the internal subscription.
+      }
+    }
+
+    let onSubscribe: () => void;
+    let onUnsubscribe: () => void;
+    let signal: TestSignal;
+
+    beforeEach(() => {
+      onSubscribe = sinon.spy();
+      onUnsubscribe = sinon.spy();
```

Comment:
Just a small improvement; I believe, the conception of `stub` fits here better.
```suggestion
      onSubscribe = sinon.stub();
      onUnsubscribe = sinon.stub();
```

Reply:
Thanks for the comment, just out of curiosity can I ask why?
To my understanding a spy is for monitoring and tracking function calls, while stub is more versatile, and I mostly use it e.g. when I want to replace/mock the implementation of a function to return something and etc.
Here, I'm only tracking the function calls and nothing more.

Reply:
The idea here is that `spy` wraps an existing function while `stub` creates a new one.

Reply:
Changed to stub.